### PR TITLE
[DOC beta] Glimmer Component & Tracking docs to target Octane release

### PR DESCRIPTION
**Pull request by jenweber**
This is ready for preliminary review, but should not be merged yet.

To-Do:
- fill in explanations for each API
- fill in args and returned values (if any)
- Give an example for each API method/property
- Give an example of the component overall in the top section
- proofread for accuracy

Contributors, please branch from `glimmer-docs` and open a PR back into this branch if you have something to add. Thanks a bunch!

Screenshot_2019-10-24  glimmer component - 3 15 - Ember API Documentation

Screen Shot 2019-10-24 at 8 57 38 PM

### CODE REVIEW

**Comment by jenweber** [comment]
This is something that still needs help from other knowledgeable people in order to get it over the finish line. @tomdale can you help out with this one?

File: packages/@ember/-internals/glimmer/lib/glimmer-component-docs.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    if (this.args.fadeIn === true) {
+      // fade in logic here
+    }
+  }
+}
+```
+*/
+
+/**
+`didInsertElement` is called when the component has been inserted into the DOM.
```
Comment (pzuraq) [inline packages/@ember/-internals/glimmer/lib/glimmer-component-docs.ts]: This is only available in Glimmer.js apps, we should not include it in the API docs for Ember. It'll be removed soon.

File: packages/@ember/-internals/glimmer/lib/glimmer-component-docs.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+@for @glimmer/component
+@public
+
+
+*/
+
+/**
+`didUpdate` is called when the component has updated and rerendered itself.
+It is called only during a rerender, not during an initial render.
+@method didUpdate
```
Comment (pzuraq) [inline packages/@ember/-internals/glimmer/lib/glimmer-component-docs.ts]: This is only available in Glimmer.js apps, we should not include it in the API docs for Ember. It'll be removed soon.

**Review by pzuraq — CHANGES_REQUESTED** [review]

File: packages/@ember/-internals/glimmer/lib/glimmer-component-docs.ts [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+  ```
+
+  Service injections and arguments are available in the constructor.
+
+  ```javascript
+  import Component from '@glimmer/component';
+  import { inject as service } from '@ember/service';
+
+  export default class SomeComponent extends Component {
+    @service animation;
```
Comment (jenweber) [inline packages/@ember/-internals/glimmer/lib/glimmer-component-docs.ts]:
```suggestion
    @service myAnimations;
```

I have found it useful to preface things that aren't built into Ember with `my` or `custom` in order to help new people who don't know where the boundaries are.

File: packages/@ember/-internals/glimmer/lib/glimmer-component-docs.ts [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+
+    constructor(owner, args) {
+      super(owner, args);
+
+      if (this.args.fadeIn === true) {
+        this.animation.register(this, 'fade-in');
```
Comment (jenweber) [inline packages/@ember/-internals/glimmer/lib/glimmer-component-docs.ts]:
```suggestion
        this.myAnimations.register(this, 'fade-in');
```

File: packages/@ember/-internals/glimmer/lib/glimmer-component-docs.ts [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+  @for @glimmer/component
+  @param {Object} owner An owner object.
+  @param {Object} args An args object.
+  @public
+*/
+
+/**
+  `willDestroy` is called after the component has been removed from the DOM, but
+  before the component is fully destroyed. This lifecycle hook can be used to
+  cleanup the component and any related state.
```
Comment (jenweber) [inline packages/@ember/-internals/glimmer/lib/glimmer-component-docs.ts]:
```suggestion
  cleanup the component and any related state, such as polling functions or websocket connections.
```

File: packages/@ember/-internals/glimmer/lib/glimmer-component-docs.ts [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+
+  ```javascript
+  import Component from '@glimmer/component';
+  import { inject as service } from '@ember/service';
+
+  export default class SomeComponent extends Component {
+    @service animation;
```
Comment (jenweber) [inline packages/@ember/-internals/glimmer/lib/glimmer-component-docs.ts]:
```suggestion
    @service myAnimations;
```

File: packages/@ember/-internals/glimmer/lib/glimmer-component-docs.ts [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+
+    willDestroy() {
+      this.animation.unregister(this);
```
Comment (jenweber) [inline packages/@ember/-internals/glimmer/lib/glimmer-component-docs.ts]:
```suggestion
      this.myAnimations.unregister(this);
```

**Review by jenweber — COMMENTED** [review]
Small stylistic suggestions.

**Comment by jenweber** [comment]
Unfortunately there is not a nice way to show an `@property` on a module. They can only go on Classes, which would add more noise to the table of contents. So, I moved all the API methods and properties into the module description for `@glimmer/component`. `tracked` is done "correctly."

Screenshot_2019-11-13  glimmer component - 3 18 - Ember API Documentation
Screenshot_2019-11-13  glimmer tracking - 3 18 - Ember API Documentation
Screenshot_2019-11-13 tracked - 3 18 - Ember API Documentation

**Review by rwjblue — APPROVED** [review]

**Review by pzuraq — APPROVED** [review]

**Pull request merged**